tests/test_environ.py

```diff
@@ -422,23 +416,65 @@ def test_register_custom_var_str(val, converted):
     assert env["MY_SPECIAL_VAR"] == converted
 
 
+def test_register_var_without_setting_val():
+    env = Env()
+    env.register("MY_SPECIAL_VAR", type="str")
+    # If variable registered but not initialized it should be None.
+    assert env["MY_SPECIAL_VAR"] == None
```

Comment:
Well, it should be the default value not, None

Reply:
And if the default is not specified it should be `DefaultNotGiven`, not sure how `None` gets in here

Reply:
Ohhh I see the confusion. I think the `default` keyword argument of register should be `DefaultNotGiven`, not `None`

Reply:
Sorry, but I would not change `None` to `DefaultNotGiven` in this PR because it's a new requirement. This change could be made in another PR.

Reply:
OK, I think that PR needs to go in first then.

Reply:
No. Because `bool(DefaultNotGiven) == True` instead of `bool(None) == False`. We should use `None` value.
If we will use `DefaultNotGiven` we can't do this:
```
${...}.register('VAR', 'path')
if $VAR and $VAR.exists():   # in case of DefaultNotGiven object we will have an exception
   ...
```